force-app/main/default/lwc/miscSharedJavaScript/__tests__/miscSharedJavaScript.test.js

```diff
@@ -0,0 +1,88 @@
+import { createElement } from 'lwc';
+import MiscSharedJavaScript from 'c/miscSharedJavaScript';
+import { getTermOptions, calculateMonthlyPayment } from 'c/mortgage';
+
+jest.mock('c/mortgage', () => {
+    return {
+        getTermOptions: jest.fn(),
+        calculateMonthlyPayment: jest.fn(),
+    };
+});
+
+describe('c-misc-shared-java-script', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Clear mocks so that every test run has a clean implementation
+        jest.clearAllMocks();
+    });
+
+    it('calculates mortgage with default values', () => {
+        // Create initial element
+        const element = createElement('c-misc-shared-java-script', {
+            is: MiscSharedJavaScript,
+        });
+        document.body.appendChild(element);
+
+        expect(getTermOptions).toHaveBeenCalled();
+
+        const lightningButtonEl = element.shadowRoot.querySelector(
+            'lightning-button',
+        );
+        lightningButtonEl.dispatchEvent(new CustomEvent('click'));
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise ends in the
+        // rejected state
+        return Promise.resolve().then(() => {
+            // Check if default values for principal, term, and rate are used for mortgage calculcation
+            expect(calculateMonthlyPayment).toHaveBeenCalledWith(200000, 30, 4);
+        });
+    });
+
+    it('calculates mortgage with custom values', () => {
+        // Create initial element
+        const element = createElement('c-misc-shared-java-script', {
+            is: MiscSharedJavaScript,
+        });
+        document.body.appendChild(element);
+
+        expect(getTermOptions).toHaveBeenCalled();
```

Comment:
Same comment as above, no need to verify this was called.

Reply:
Removed